common/helpers/move/get-update-urls.js

```diff
@@ -1,29 +1,16 @@
-const getMoveUrl = require('./get-move-url')
-
-const getUpdateUrls = (move, canAccess = () => false, updateSteps = []) => {
-  const updateUrls = {}
-
-  if (move._hasLeftCustody) {
-    return updateUrls
-  }
+const editSteps = require('../../../app/move/app/edit/steps')
 
-  if (!canAccess(`move:update:${move.move_type}`)) {
-    return updateUrls
-  }
-
-  updateSteps.forEach(updateJourney => {
-    if (!canAccess(updateJourney.permission)) {
-      return
-    }
+const getMoveUrl = require('./get-move-url')
 
-    const steps = updateJourney.steps
-    const entryPointUrl = Object.keys(steps).filter(
-      step => steps[step].entryPoint
-    )[0]
-    const key = updateJourney.key
-    updateUrls[key] = getMoveUrl(move.id, 'update', { entryPointUrl })
-  })
-  return updateUrls
+function getUpdateUrls(move = {}, canAccess = () => false) {
+  return editSteps
+    .filter(() => move._canEdit)
+    .filter(step => canAccess(step.permission))
+    .reduce((acc, current) => {
+      const entryPointUrl = Object.keys(current.steps)[0]
+      acc[current.key] = getMoveUrl(move.id, 'update', { entryPointUrl })
+      return acc
+    }, {})
```

Comment:
```suggestion
  return Object.fromEntries(
    editSteps
      .filter(() => move._canEdit)
      .filter(step => canAccess(step.permission))
      .map(current => {
        const entryPointUrl = Object.keys(current.steps)[0]
        return [current.key, getMoveUrl(move.id, 'update', { entryPointUrl })]
      })
  )
```

I tend to prefer `map` over `reduce` even for objects as I find it easier to understand, but happy with either.

Reply:
Yes, that is neater actually. I tend to prefer reading `map`s too but I initially thought it wouldn't be as nice to read transforming an array into an Object but this does read nicely too :)